contracts/src/TeleporterTokenDestination.sol

```diff
@@ -541,6 +346,215 @@ abstract contract TeleporterTokenDestination is
         return (amount, primaryFee);
     }
 
+    /**
+     * @dev Sends tokens to the specified destination token bridge instance.
+     *
+     * Requirements:
+     *
+     * - {input.destinationBridgeAddress} cannot be the zero address
+     * - {input.recipient} cannot be the zero address
+     * - {amount} must be greater than 0
+     */
+    function _processSend(SendTokensInput calldata input, uint256 amount) private {
+        _validateSingleHopInput(
+            input.destinationBridgeAddress, input.secondaryFee, input.multiHopFallback
+        );
+
+        uint256 primaryFee;
+        (amount, primaryFee) = _prepareSend({
+            amount: amount,
+            primaryFeeTokenAddress: input.primaryFeeTokenAddress,
+            primaryFee: input.primaryFee,
+            secondaryFee: input.secondaryFee
+        });
+
+        BridgeMessage memory message = BridgeMessage({
+            messageType: BridgeMessageType.SINGLE_HOP_SEND,
+            payload: abi.encode(SingleHopSendMessage({recipient: input.recipient, amount: amount}))
+        });
+
+        bytes32 messageID = _sendTeleporterMessage(
+            TeleporterMessageInput({
+                destinationBlockchainID: sourceBlockchainID,
+                destinationAddress: tokenSourceAddress,
+                feeInfo: TeleporterFeeInfo({
+                    feeTokenAddress: input.primaryFeeTokenAddress,
+                    amount: primaryFee
+                }),
+                requiredGasLimit: input.requiredGasLimit,
+                allowedRelayerAddresses: new address[](0),
+                message: abi.encode(message)
+            })
+        );
+
+        emit TokensSent(messageID, _msgSender(), input, amount);
+    }
+
+    /**
+     * @dev Sends tokens to the specified destination token bridge instance via multi-hop.
+     *
+     * Requirements:
+     *
+     * - {input.destinationBridgeAddress} cannot be the zero address
+     * - {input.recipient} cannot be the zero address
+     * - {amount} must be greater than 0
+     */
+    function _processSendMultiHop(SendTokensInput calldata input, uint256 amount) private {
+        _validateMultiHopInput(
+            input.destinationBlockchainID, input.destinationBridgeAddress, input.multiHopFallback
+        );
+
+        uint256 primaryFee;
+        (amount, primaryFee) = _prepareSend({
```

Comment:
Is this var declared here in order to reuse `amount`?

Reply:
Yeah